### PR TITLE
Support synthetic decorations in worksheets

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, we would have a separate type of decorations for worksheets. Now, we have a separate type for worksheets evaluations and synthetic decorations.

Needed for https://github.com/scalameta/metals/pull/3582

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -1068,9 +1066,6 @@ function launchMetals(
       scalaDebugger.initialize(outputChannel).forEach((disposable) => {
         context.subscriptions.push(disposable);
       });
-      client.onNotification(DecorationTypeDidChange.type, (options) => {
```
Comment (tgodzik) [inline src/extension.ts:1071]: This is never used and not sure if it actually should.

File: src/extension.ts [hunk 2]
```diff
@@ -1068,9 +1066,6 @@ function launchMetals(
...
-        decorationType = window.createTextEditorDecorationType(options);
-      });
       client.onNotification(DecorationsRangesDidChange.type, (params) => {
         const editors = window.visibleTextEditors;
         const path = Uri.parse(params.uri).toString();
```
Comment (kpodsiad) [inline src/extension.ts:1071]: I've noticed that `client.onNotification` return `Disposable` but we're discarding returned value. Maybe it should be added to `context.subscriptions` (yeah, it's a broader problem)

`extension.ts` is a very large file, I wonder if we can e.g. create `notifications` dir and extract all `client.onNotification` to different files

**Review by kpodsiad — APPROVED** [review]
I'd love to see some refactor which will reduce `extension.ts` size, but it isn't obligatory ;)

File: src/extension.ts (hunk 2 above)
Comment (tgodzik) [reply src/extension.ts]: I added all the disposables from `onNotification` to subscriptions. Good idea!

Comment (kpodsiad) [reply src/extension.ts]: Thanks, that's a step into good direction ;)

**Review by kpodsiad — APPROVED** [review]

**Pull request merged**